core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -2783,6 +2846,10 @@ private RecordCursorFactory generateSampleBy(
         final ExpressionNode offset = model.getSampleByOffset();
         final Function offsetFunc;
         final int offsetFuncPos;
+        Function sampleFromFunc = null;
```

Comment:
nit: initialization is redundant (same with `sampleToFunc`):
```suggestion
        Function sampleFromFunc;
```